src/flash/nor/rp2040.c

```diff
@@ -172,16 +210,7 @@ static int rp2040_flash_write(struct flash_bank *bank, const uint8_t *buffer, ui
 		return ERROR_TARGET_UNALIGNED_ACCESS;
 	}
 
-	LOG_DEBUG("Connecting internal flash");
-	err = rp2040_call_rom_func(bank->target, priv->stacktop, FUNC_CONNECT_INTERNAL_FLASH, NULL, 0);
-	if (err != ERROR_OK)
-	{
-		LOG_ERROR("RP2040 write: failed to connect internal flash");
-		return err;
-	}
-
-	LOG_DEBUG("Kicking flash out of XIP mode");
-	err = rp2040_call_rom_func(bank->target, priv->stacktop, FUNC_FLASH_EXIT_XIP, NULL, 0);
+	err = rp2040_flash_exit_xip (bank);
 	if (err != ERROR_OK)
 	{
 		LOG_ERROR("RP2040 write: failed to exit flash XIP mode");
```

Comment:
Probably not much point in doing `LOG_ERROR` in both `rp2040_flash_exit_xip` **and** `rp2040_flash_write` ?
I.e. this should probably be just:
```
	err = rp2040_flash_exit_xip(bank);
	if (err != ERROR_OK)
		return err;
```
:shrug: 
(also, please try to stick to the existing formatting-style of no space between `rp2040_flash_exit_xip` and `(bank)`)

Reply:
Done